angular-workspace/projects/ni/nimble-angular/pipes/format-number-text.pipe.ts

```diff
@@ -0,0 +1,72 @@
+import { Inject, LOCALE_ID, Pipe, PipeTransform } from '@angular/core';
+import { NumberTextFormat } from '@ni/nimble-components/dist/esm/table-column/number-text/types';
+import { NumberTextUnitFormat } from '@ni/nimble-components/dist/esm/table-column/number-text/models/number-text-unit-format';
+import { passthroughUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/passthrough-unit-scale';
+import { byteUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-unit-scale';
+import { byte1024UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-1024-unit-scale';
+import { voltUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/volt-unit-scale';
+import type { UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/unit-scale';
+
+export { byteUnitScale, byte1024UnitScale, voltUnitScale };
+
+/**
+ * A pipe that transforms a number into a string with optional unit
+ */
+@Pipe({
+    name: 'formatNumberText',
+    standalone: true
+})
+export class FormatNumberTextPipe implements PipeTransform {
+    /**
+     * @internal
+     */
+    public numberTextUnitFormat?: NumberTextUnitFormat;
+    private numberTextFormat?: NumberTextFormat;
+    private decimalDigits?: number;
+    private decimalMaximumDigits?: number;
+    private unitScale?: UnitScale;
+
+    public constructor(@Inject(LOCALE_ID) private readonly locale: string) {}
+
+    public transform(value: number, {
+        numberTextFormat = NumberTextFormat.default,
+        decimalDigits,
+        decimalMaximumDigits,
+        unitScale = passthroughUnitScale
+    }: {
+        numberTextFormat: 'decimal',
```

Comment:
I get a compile error if I try to use `NumberTextFormat.decimal` here instead of the value literal. I think the benefit of having type errors for incompatible options outweighs the downside of using `'decimal'` and `undefined` here instead of `NumberTextFormat.default/decimal`.